### PR TITLE
Use mixxx::audio::SampleRate when dealing with beats

### PR DESCRIPTION
**Pull request by daschuer**

### CODE REVIEW

File: src/track/beatfactory.cpp [hunk 1]
```diff
@@ -26,24 +26,26 @@ mixxx::BeatsPointer BeatFactory::loadBeatsFromByteArray(
 }
 
 mixxx::BeatsPointer BeatFactory::makeBeatGrid(
-        SINT sampleRate, double dBpm, double dFirstBeatSample) {
+        mixxx::audio::SampleRate sampleRate,
+        double dBpm,
```
Comment (Holzhaus) [inline src/track/beatfactory.cpp:30]: We also have `mixxx::Bpm`, but I think #2961 already uses that.

File: src/track/beatfactory.cpp [hunk 2]
```diff
@@ -92,29 +94,32 @@ QString BeatFactory::getPreferredSubVersion(
...
-        const bool bEnableFixedTempoCorrection,
-        const bool bEnableOffsetCorrection,
-        const int iSampleRate,
-        const int iTotalSamples,
-        const int iMinBpm,
-        const int iMaxBpm) {
+        bool bEnableFixedTempoCorrection,
+        bool bEnableOffsetCorrection,
+        mixxx::audio::SampleRate sampleRate,
+        SINT totalSamples,
```
Comment (Holzhaus) [inline src/track/beatfactory.cpp:100]: Are these mono samples or stereo samples (i.e. frames)? If the latter, let's use `totalFrames` instead.

**Review by Holzhaus — COMMENTED** [review]
Great, thanks.

Unfortunately, there are some failing tests now:

```
	 38 - BeatMapTest.Scale (Failed)
	 39 - BeatMapTest.TestNthBeat (Failed)
	 40 - BeatMapTest.TestNthBeatWhenOnBeat (Failed)
	 41 - BeatMapTest.TestNthBeatWhenOnBeat_BeforeEpsilon (Failed)
	 42 - BeatMapTest.TestNthBeatWhenOnBeat_AfterEpsilon (Failed)
	 43 - BeatMapTest.TestNthBeatWhenNotOnBeat (Failed)
	 44 - BeatMapTest.TestBpmAround (Failed)
```

File: src/track/beatfactory.cpp (hunk 2 above)
Comment (daschuer) [reply src/track/beatfactory.cpp:100]: It are frames * 2 in this case.

Comment (Holzhaus) [reply src/track/beatfactory.cpp:100]: Ok, than let's leave it as-is.

**Review by Holzhaus — APPROVED** [review]
LGTM. thank you. I'll use Squash-and-Merge to keep bisectability on windows.

**Pull request merged**